core/unit_test/CMakeLists.txt

```diff
@@ -287,6 +287,7 @@ KOKKOS_ADD_TEST_EXECUTABLE(
 )
 # We need -rdynamic on GNU platforms for the stacktrace functionality
 # to work correctly with shared libraries
+if(NOT KOKKOS_HAS_TRILINOS)
 SET_PROPERTY(TARGET StackTraceTestExec PROPERTY ENABLE_EXPORTS 1)
```

Comment:
The issue is that the exe target name with TriBITS.  You should have added a comment.